src/Components/WebAssembly/WebAssembly/src/Services/LazyAssemblyLoader.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using System.Runtime.InteropServices;
+using System.Runtime.Loader;
+using Microsoft.JSInterop.WebAssembly;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Services
+{
+    public class LazyAssemblyLoader
```

Comment:
Needs XML docs

Reply:
Should we call this DeferredAssemblyLoader \ AsyncAssemblyLoader?

Reply:
I've added the XML docs.

> Should we call this DeferredAssemblyLoader \ AsyncAssemblyLoader?

I named it lazy for the consistency with the property group that we added to the project file as part of the last PR. It's probably the most explicit and well-defined name of the bunch that we've considered so far.